blog/2024-07-30-how-to-run-webassembly-components-on-kubernetes-with-wasmcloud/index.mdx

```diff
@@ -0,0 +1,178 @@
+:::
+
+A wasmCloud installation requires a few things to run:
+
+- **[NATS](/docs/ecosystem/nats/)** (with Jetstream enabled)
+- **[wasmCloud Application Deployment Manager (wadm)](/docs/ecosystem/wadm/)** connected to NATS
+
+If you're not sure what that means, don't worry: the following steps will help you deploy the prerequisites to your Kubernetes cluster, then deploy the operator.
+
+#### 1. Deploy prerequisites
+
+First we'll add the NATS Helm repo. If you're not familiar with it, NATS is the open source connective technology that wasmCloud uses to create a unified topology across any number of environments. 
+
+```shell
+helm repo add nats https://nats-io.github.io/k8s/helm/charts/
+```
+Next install the upstream NATS Helm chart:
+
+```shell
+helm upgrade --install -f https://raw.githubusercontent.com/wasmCloud/wasmcloud-operator/main/examples/quickstart/nats-values.yaml nats nats/nats
+```
+Deploy wadm to your Kubernetes cluster with a Helm chart hosted by wasmCloud. 
+
+```shell
+helm install wadm -f https://raw.githubusercontent.com/wasmCloud/wasmcloud-operator/main/examples/quickstart/wadm-values.yaml oci://ghcr.io/wasmcloud/charts/wadm
+```
+You can validate the installations with:
+
+```shell
+kubectl rollout status deploy,sts -l app.kubernetes.io/instance=nats
+kubectl rollout status deploy -l app.kubernetes.io/instance=wadm
+```
+Once NATS and wadm are ready, you can move on to deploy the operator.
+
+#### 2. Deploy the operator
+
+To deploy the operator:
+
+```shell
+kubectl apply -k https://github.com/wasmCloud/wasmcloud-operator/deploy/base
+```
+Validate the installation by checking that the pods are ready...
+
+```shell
+kubectl rollout status deploy -l app=wasmcloud-operator -n wasmcloud-operator
+```
+...and confirming that the API service is available:
+```shell
+kubectl get apiservices.apiregistration.k8s.io v1beta1.core.oam.dev
+```
+
+#### 3. Create wasmCloud hosts
+
+Our final step is to apply the `wasmcloud-host` manifest:
+
+```shell
+kubectl apply -f https://raw.githubusercontent.com/wasmCloud/wasmcloud-operator/main/examples/quickstart/wasmcloud-host.yaml
+```
+Check the status of your wasmCloud host:
+
+```shell
+kubectl describe wasmcloudhostconfig wasmcloud-host
+```
+
+At this point, you're up and running. Let's try running some WebAssembly components with our new wasmCloud installation. 
+
+#### Run a WebAssembly component on Kubernetes
+
+When you `kubectl apply` a [wasmCloud application manifest](../../docs/ecosystem/wadm/model), the cluster will automatically provision the component workload with wasmCloud. We'll use the [`hello-world-application.yaml` manifest](https://raw.githubusercontent.com/wasmCloud/wasmcloud-operator/main/examples/quickstart/hello-world-application.yaml) included in the operator's quickstart.
```

Comment:
It might be nice to mention/link to the application that's being deployed in the `wasmcloud/wasmcloud` repository, so that in case someone is curious (or suspicious) as to what the application is that we're deploying, they can see the source code: https://github.com/wasmCloud/wasmCloud/tree/main/examples/rust/components/http-hello-world